experiment/build/generate_cloudbuild.py

```diff
@@ -45,32 +85,59 @@ def create_cloud_build_spec(buildable_images, docker_registry):
         if 'depends_on' in image:
             step['wait_for'] = []
             for dep in image['depends_on']:
-                step['wait_for'] += [dep]
-        cloud_build_spec['images'].append(name)
+                step['wait_for'] += [_identity(dep)]
+        image_built = EXPERIMENT_TAG + '/' + name + ':' + experiment_var
+        cloud_build_spec['images'].append(image_built)
         cloud_build_spec['steps'].append(step)
 
+    if coverage:
+        cloud_build_spec['steps'] += coverage_steps()
+
     return cloud_build_spec
 
 
-def main():
+def generate_base_images(buildable_images):
+    base_images_template = {}
+    for name in buildable_images:
+        if 'base' in name:
+            base_images_template[name] = buildable_images[name]
+    return create_cloud_build_spec(base_images_template)
+
+
+def generate_benchmark_images(buildable_images, coverage=False):
+    benchmark_images_template = {}
+    for name in buildable_images:
+        if any(_ in name for _ in ('base', 'oss-fuzz')):
+            continue
+        benchmark_images_template[name] = buildable_images[name]
+    return create_cloud_build_spec(benchmark_images_template, coverage=coverage)
+
+
+def generate_oss_fuzz_benchmark_images(buildable_images, coverage=False):
+    oss_fuzz_benchmark_images_template = {}
+    for name in buildable_images:
+        if 'oss-fuzz' in name:
+            oss_fuzz_benchmark_images_template[name] = buildable_images[name]
+    return create_cloud_build_spec(oss_fuzz_benchmark_images_template, coverage=coverage)
+
+
+def write_build_spec(filename, data):
+    file_path = os.path.join(os.path.dirname(__file__), os.pardir, os.pardir,
+                            'docker', 'gcb', filename)
+    yaml_utils.write(file_path, data)
+
+
+def generate_gcb_build_spec():
```

Comment:
Would it be possible to create a single spec (ie. one yaml/json file) that contains everything?

Reply:
As far as I understand, it should be possible but that would require a different change such as using GCB API calls instead of using the command that we do currently. So using the API, specific steps could be selected from one .yaml file and then passed to API.

Reply:
Sorry, I don't understand what does the API has to do with this. Currently, you write a bunch of specs separately (base-images.yaml, fuzzers,yaml, coverage.yaml, .....). Can you just create a single one that contains everything?

Reply:
As per my understanding, the current implementation calls a command using `new_process.execute` and passes the config file (like `base-images.yaml`) to the gcloud command. So if the spec is put in one file, the config file argument couldn't be passed. So to select specific steps from a single file, I think the steps need to be passed as an object to the API call.

Reply:
right, but do we need to build these things separately? i.e. can we have a single spec (eg all-images-we-need.yaml), and call gcloud build submit . --config all-images-we-need.yaml?

Reply:
I guess --config all_images.yaml would look at building all images for given set of substitutions (includes _REPO, _FUZZER, _BENCHMARK, _EXPERIMENT), but having separate files helps build only, say coverage benchmark images. As far as I can tell, the only way to combine all into one spec file is to change the build code to use API instead of command.